elki/src/main/java/ch/ethz/globis/pht/util/Refs.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2011-2015 ETH Zurich. All Rights Reserved.
+ *
+ * This software is the proprietary information of ETH Zurich.
+ * Use is subject to license terms.
+ */
+package ch.ethz.globis.pht.util;
+
+import java.util.Arrays;
+
+import ch.ethz.globis.pht.PhTreeHelper;
+
+
+public class Refs {
+	
+	private static final int BYTES_PER_UNIT = 4;//bytes
+//	private static int statAExpand;
+//	private static int statACreate;
+//	private static int statATrim;
+	
+	public static final Object[] EMPTY_REF_ARRAY = {};
+    private static final ArrayPool POOL = 
+    		new ArrayPool(PhTreeHelper.ARRAY_POOLING_MAX_ARRAY_SIZE, 
+    				PhTreeHelper.ARRAY_POOLING_POOL_SIZE);
+
+    
+    private static class ArrayPool {
```

Comment:
Array pooling is an interesting function. It may be a bit unfair though to only have this available in the PhTree, not in the other indexes, which could also benefit from object pooling and recycling. :-(